### PR TITLE
Fix(Headless, Quantic): Sending the correct analytics event when clicking on the title of the Quickview of a document suggestion

### PR DESCRIPTION
**Pull request by mmitiche**
### Changes made in Headless:
- New controller `buildCaseAssistInteractiveResult` created. this controller will take care of sending the correct analytics events when the user clicks on the title of a document suggestion in a quickview.
- New analytic action created `logDocumentSuggestionOpen`, this action dispatch a `suggestion_click` event with metadata indicating that the click was made to open the document suggestion after clicking on its title from the quickview.

### Changes made in Quantic:
- The QuanticResultLink calls the  buildCaseAssistInteractiveResult controller when being used in a Case Assist use case.
- The value of target in the QuanticResultLink  when used in the QuanticResultQuickview updated to be "_blank" instead of "_self", to open the link in a new tab instead of opening it in the same tab.

https://user-images.githubusercontent.com/86681870/154520589-127b9dd1-5af4-45d8-9159-6cee53bc750f.mov

### CODE REVIEW

**Review by btaillon — APPROVED** [review]
looks good to me

**Review by ThibodeauJF — APPROVED** [review]

File: packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.html [hunk 1]
```diff
@@ -3,10 +3,6 @@
     <div class="error-message slds-text-color_destructive">{renderingError}</div>
   </template>
   <template if:false={renderingError}>
-    <template if:true={showQuickview}>
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.html:6]: yes!!!

File: packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.js [hunk 2]
```diff
@@ -30,13 +30,6 @@ export default class QuanticDocumentSuggestion extends LightningElement {
    * @type {string}
    */
   @api engineId;
-  /**
-   * The ID of the search engine instance the component registers to, this is used to instantiate the search interface to be able to show the quick view.
-   * @api
-   * @type {string}
-   * @defaultValue `'search-engine'`
-   */
-  @api searchEngineId = 'search-engine';
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.js:39]: awesome

File: packages/quantic/force-app/main/default/lwc/quanticResultLink/quanticResultLink.js [hunk 3]
```diff
@@ -33,9 +33,16 @@ export default class QuanticResultLink extends LightningElement {
    *   - `_top`: the topmost browsing context (the "highest" context that’s an ancestor of the current one). If there are no ancestors, this behaves as `_self`.
    * @api
    * @type {string}
-   * @defaultValue `'_self'`
+   * @defaultValue `'_blank'`
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticResultLink/quanticResultLink.js:36]: Uhm idk if we should change the default value in the resultLink component. Why not just set the `target` option to "_blank" from the quickview component that uses it?

**Review by nathanlb — COMMENTED** [review]
Looks great! Love that we remove the search engine from case-assist document suggestion! 🙏 
Not sure we need to change the default on the base resultlink component though. The target=_self default is kind of a staple search page behaviour.

**Pull request merged**